### ISSUE TITLE
Custom Headers

### ISSUE THREAD

**Issue by Radiergummi** [joseluisq/static-web-server#100]
**Is your feature request related to a problem? Please describe.**  
As mentioned in [#30](https://github.com/joseluisq/static-web-server/issues/30#issuecomment-1104922015), I would like to customise the security headers to add a nonce or hash to my CSP policy.

**Describe the solution you'd like**  
The easiest way to solve this would be a new configuration option to add custom headers. This flag should be repeatable on the command line, and support line breaks in environment variables.
It could be used like this:
```bash
static-web-server --header "Content-Security-Policy: script-src 'strict-dynamic' 'nonce-R4nd0m'"

# Shorthand -h
static-web-server -h "Foo: bar"

# Repeated usage to add multiple headers
static-web-server -h "Foo: bar" -h "Foo: baz" -h "Test: 42"
```

This would also allow users to quickly specify a custom cache control directive, for example. There are lots of other use cases where static headers are sensible and useful.

**Describe alternatives you've considered**  
There's not many - passing on a locked-down CSP, or placing SWS behind an nginx instance, or adding a Traefik middleware; but all of those are neither desirable nor feasible in some environments.

**Comment by joseluisq** [comment]
The idea in general is not bad, since having something similar to `curl` via its `-H, --header` (CLI) could help users to customize headers easily.

In addition, I plan to support a config file for more advanced use cases which includes headers customization.
However until then, this feature request could be a good progress forward.

**Comment by joseluisq** [comment]
BTW this feature request could help to resolve #30.

**Comment by joseluisq** [comment]
Custom HTTP headers will be available in next `v2.8.0` via the new config file support. See PR for more details.

> The idea in general is not bad, since having something similar to `curl` via its `-H, --header` (CLI) could help users to customize headers easily.
> 
> In addition, I plan to support a config file for more advanced use cases which includes headers customization. However until then, this feature request could be a good progress forward.

However the **CLI** idea could be fine for global headers and in a similar way like `curl` does it. But we could revisit that idea in near future.

But for now this feature request is addressed by #101.

**Comment by joseluisq** [comment]
Released on [v2.8.0](https://github.com/joseluisq/static-web-server/releases/tag/v2.8.0)
